firebase_admin/firestore.py

```diff
@@ -50,15 +51,22 @@ def client(app=None):
     .. _Firestore Client: https://googlecloudplatform.github.io/google-cloud-python/latest\
           /firestore/client.html
     """
-    fs_client = _utils.get_app_service(app, _FIRESTORE_ATTRIBUTE, _FirestoreClient.from_app)
+
+    options = {"database_id": database_id} if database_id else None
+
+    fs_client = _utils.get_app_service(app, options, _FIRESTORE_ATTRIBUTE, _FirestoreClient.from_app)
     return fs_client.get()
 
 
 class _FirestoreClient:
     """Holds a Google Cloud Firestore client instance."""
 
-    def __init__(self, credentials, project):
-        self._client = firestore.Client(credentials=credentials, project=project)
+    def __init__(self, credentials, project, database_id=None):
+        if database_id:
+            self._client = firestore.Client(credentials=credentials, project=project, database=database_id)
```

Comment:
note: the `database` parameter is only upported in firestore [v2.12.0+](https://github.com/googleapis/python-firestore/releases/tag/v2.12.0), but this repo supports [v2.9.1](https://github.com/firebase/firebase-admin-python/blob/b992604fb831f19a91430bc699946323e86fb344/setup.py#L43). 

It's probably ok, but we might want to double-check that it fails gracefully when used with an older version

Reply:
Wasn't sure the best way to do this but I included a version check if attempting to use older that v2.12.0

Reply:
I think we can update the firestore version to the latest in the next release so this check probably won't be necessary (we don't check the version in other SDKs). I understand this PR has been sitting here for a while so I will first merge this and then remove the version check in a follow up PR :)